web/src/stream_list.ts

```diff
@@ -967,7 +969,7 @@ function on_sidebar_channel_click(
     }
 
     const section_for_stream = stream_list_sort.current_section_id_for_stream(stream_id);
-    if (collapsed_sections.has(section_for_stream)) {
+    if (section_for_stream !== undefined && collapsed_sections.has(section_for_stream)) {
```

Comment:
I think it might not be possible for this to be `undefined` today, but we may have future UI that shows non-subscribed channels in the sidebar when you're previewing that channel, so I think this code is correct.